### PR TITLE
Refactor/collection explorer/folder explorer

### PR DESCRIPTION
**Pull request by theketan26**
## Description

Refactor collection explorer page, issue: #778 
Refactor folder explorer page, issue: #786 

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md

### CODE REVIEW

**Comment by punitkashyup** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @theketan26 on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:white_check_mark: OK|1|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|

[View on SonarQube](https://sonar.techdomeaks.com/dashboard?id=sparrow)
###### _updated: 5/1/2024, 13:57:51 (UTC+0)_

**Comment by punitkashyup** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @theketan26 on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:white_check_mark: OK|1|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|

[View on SonarQube](https://sonar.techdomeaks.com/dashboard?id=sparrow)
###### _updated: 5/2/2024, 06:06:56 (UTC+0)_

**Comment by punitkashyup** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @theketan26 on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:white_check_mark: OK|1|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|

[View on SonarQube](https://sonar.techdomeaks.com/dashboard?id=sparrow)
###### _updated: 5/2/2024, 06:14:08 (UTC+0)_

File: src/lib/utils/helpers/handle-clicks.helper.ts [hunk 1]
```diff
@@ -1,94 +1,94 @@
...
-  });
+// /* eslint-disable @typescript-eslint/no-explicit-any */
+// import { isFolderCreatedFirstTime } from "$lib/store/collection";
+// import { CollectionsViewModel } from "../../../pages/Collections/Collections.ViewModel.old";
+// import { ItemType } from "../enums/item-type.enum";
+// import type { Path } from "../interfaces/request.interface";
+// import { generateSampleCollection } from "../sample/collection.sample";
+// import { generateSampleFolder } from "../sample/folder.sample";
+// import { generateSampleRequest } from "../sample/request.sample";
+// import { moveNavigation } from "./navigation";
```
Comment (Astitva877) [inline src/lib/utils/helpers/handle-clicks.helper.ts:9]: Why this whole file is commented?

**Review by itsmdasifraza — APPROVED** [review]

File: src/lib/utils/helpers/handle-clicks.helper.ts (hunk 1 above)
Comment (theketan26) [reply src/lib/utils/helpers/handle-clicks.helper.ts:9]: This file is not being used in code, all necessary functionalities from this file are transferred to CollectionsPage.ViewModel.ts. This is here only for reference for now.

**Comment by punitkashyup** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @theketan26 on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:white_check_mark: OK|1|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|

[View on SonarQube](https://sonar.techdomeaks.com/dashboard?id=sparrow)
###### _updated: 5/8/2024, 05:34:34 (UTC+0)_

**Comment by theketan26** [comment]
@gc-codes, merge conflicts resolved please review the PR.

**Review by gc-codes — APPROVED** [review]

**Pull request merged**